source/braille.py

```diff
@@ -1974,8 +1990,23 @@ def _get_scriptableObject(self):
 			return display
 		return super(BrailleDisplayGesture, self).scriptableObject
 
+	#: Compiled regular expression to match an identifier 
+	#: including an optional model name
+	#: @type: RegexObject
+	ID_PARTS_REGEX = re.compile(r"br\((\w+)(\.(\w+))?\):([\w+]+)", re.U)
+
 	@classmethod
 	def getDisplayTextForIdentifier(cls, identifier):
-		return handler.display.description, identifier.split(":", 1)[1]
+		idParts = cls.ID_PARTS_REGEX.search(identifier)
+		if not idParts:
+			raise ValueError("Invalid identifier provided: %s"%identifier)

```

Comment:
I have the impression the old version of the function did not expect malformed input (despite in case of malformed input it may also have ended up in IndexErrors).
Raising an exception in case it is needed is good, but where will it be catched and handled in an appropriate way?

Reply:
This exception is raised to give a readable clarification of what is going wrong, if there is at all something going wrong. The old version of this function indeed didn't expect mallformed input, and I think this is a safe assumption. Personally I belief we shouldn't catch this, as when this is raised, it would be a quite serious error.

Reply:
Would an assert be more appropriate in that case?
At least from an intention point of view, it may resemble more the implicit contract of the original version of the function "never ever supply malformed input, otherwise you are in deep trouble" versus "in case you supply malformed input (please don't), I will be so kind to let you know".
Of course, in the end the effect would be the same, as python turns a failing assert also into an exception, in contrast to other languages where an assert triggers a pretty abrupt emergency break.

Reply:
> Would an assert be more appropriate in that case?

Assertions get removed in release builds, so handling this with an assertion statement will make no difference to the original assumption that no malformed input will be supplied to this function. I can live with an assertion though.

@dkager @michaeldcurran, thoughts?